feat(web-modeler): add support for configuring Zeebe clusters

jfriedenstab opened the pull request:
### Which problem does the PR fix?
Closes https://github.com/camunda/web-modeler/issues/10774.

Closes #2478 

### What's in this PR?
Adds support for configuring Zeebe clusters that will be available in Web Modeler to deploy resources to.
If Web Modeler is installed together with Zeebe in the same Helm installation, this Zeebe cluster will be pre-configured.

<!--
  Explain the contents of the PR.
  Give an overview of the implementation, which decisions were made, and why.
-->

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/docs/contributing.md).

<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->

**Before opening the PR:**

- [x] In the repo's root dir, run `make go.update-golden-only`.
- [x] There is no other open [pull request](../pulls) for the same update/change.
- [x] Tests for charts are added (if needed).
- [x] In-repo [documentation](../blob/main/docs/contributing.md#documentation) are updated (if needed).

**After opening the PR:**

- [x] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [x] Did all checks/tests pass in the PR?

<!--
### To-Do

- [ ] If the PR is not complete but you want to discuss the approach,
  list what remains to be done here.
-->

jfriedenstab commented on charts/camunda-platform-alpha/templates/camunda/_helpers.tpl:
```diff
@@ -539,7 +550,7 @@ Zeebe templates.
     {{-
       printf "http://%s:%v%s"
-        (include "zeebe.fullname.gateway" .)
+        (include "zeebe.names.gateway" .)
```
FYI: I changed this so that `.Values.global.zeebeClusterName` is considered when creating the URL.

jessesimpson36 replied:
I don't see any problems with this but thanks for pointing it out.

jfriedenstab commented on charts/camunda-platform-alpha/templates/camunda/_helpers.tpl:
```diff
@@ -364,10 +364,10 @@ Usage: {{ include "camundaPlatform.getExternalURL" (dict "component" "operate" "
     {{- if (index .context.Values .component "ingress" "enabled") }}
       {{- $proto := ternary "https" "http" (index .context.Values .component "ingress" "tls" "enabled") -}}
-      {{- printf "%s://%s" $proto (index .context.Values .component "ingress" "host") -}} 
+      {{- printf "%s://%s" $proto (index .context.Values .component "ingress" "host") -}}
```
FYI: The trailing whitespaces were removed automatically by my IDE.

jessesimpson36 commented:
I did two manual tests:
1. I installed the helm chart using a combined ingress and deployed+created a process instance using the simplified form
2. I installed the helm chart using Azure OIDC and deployed + created a process instance using the simplified form.

Both of those seem to work.  I also tried clearing out some of the urls inside the configmap to see how the web modeler handles improper operate+tasklist urls (for situations where those may be disabled), and it seems to handle those situations gracefully.

jessesimpson36 commented:
I think the QA team's E2E tests will need to be updated to reflect these changes to the UI. So I would say lets hold on merging this until we get 1 successful nightly run (we've seen a lot of failures lately),  and then we can merge this, and then update the QA tests after.

Now I'm going to check on why the integration tests are failing.

jfriedenstab commented:
> Now I'm going to check on why the integration tests are failing.

@jessesimpson36 The integration tests have passed now ✅. I rebased the branch on `main` – not sure if that was the reason 🙂.
By the way, I've also added an additional `id` field to the cluster configuration (related issue: https://github.com/camunda/web-modeler/issues/11401).

jfriedenstab commented:
Regarding:
> I think the QA team's E2E tests will need to be updated to reflect these changes to the UI. So I would say lets hold on merging this until we get 1 successful nightly run (we've seen a lot of failures lately), and then we can merge this, and then update the QA tests after.

I've quickly created a PR to adapt the E2E tests: https://github.com/camunda/c8-cross-component-e2e-tests/pull/400 (cc @slolatte).

jessesimpson36 commented:
thank you @jfriedenstab , I'm super excited about this change.

jessesimpson36 commented:
Merge when you're ready
